src/core/public/application/types.ts

```diff
@@ -740,6 +740,26 @@ export interface NavigateToAppOptions {
    * if true, will open the app in new tab, will share session information via window.open if base
    */
   openInNewTab?: boolean;
+
+  /**
+   * if true, will bypass the default onAppLeave behavior
+   */
+  skipAppLeave?: boolean;
+}
+
+/**
+ * Options for the {@link ApplicationStart.navigateToUrl | navigateToUrl API}
+ * @public
+ */
+export interface NavigateToUrlOptions {
+  /**
+   * if true, will bypass the default onAppLeave behavior
+   */
+  skipAppLeave?: boolean;
+  /**
+   * if true, will redirect directly to the url
```

Comment:
nit: maybe explain this a bit more, i.e. will redirect without checking deeplinks, etc (just so we are clear what we are forcing it to do and how it differs from skipAppLeave)

Reply:
++ to be a bit more explicit, I think we need to explain that this option will perform a forceful page assign/reload/refresh, (we could even be mentioning that effectively this means using `location.assign` instead of pushing state)

Reply:
👍 , being explicit makes perfect sense!